beacon-chain/blockchain/block_processing.go

```diff
@@ -80,6 +83,20 @@ func (c *ChainService) ReceiveBlock(ctx context.Context, block *pb.BeaconBlock)
 		return beaconState, fmt.Errorf("could not process block deposits, attestations, and other operations: %v", err)
 	}
 
+	// Check state root
```

Comment:
Actually, don't you think this goes in VerifyBlockValidity?

Reply:
It at least needs to go before cleanup block operations.

Reply:
The check moved above to CleanupBlockOperations.

In the specification, this is a separate step that requires a full state calculation, so I don’t think it’s right to include this in VerifyBlockValidity.